doc/TUTORIAL.md

```diff
@@ -54,177 +46,197 @@ This information allows other humans to help and build on your work.
-   The `url` is created to query the API.
-5. Create the basic badge to use.
-   You can read about the [different formats available][format].
-   It contains the format for all responses, regardless of the API's response.
-6. We request the `url` and pass a call back function to the request.
-   The function is called once the data is retrieved from the API.
-7. We want to always see a badge regardless the input.
-   In some cases the API may return an error or a HTTP status code indicating
-   a client error or a server error e.g. if the query was invalid. The error response
-   is handled by the [checkErrorResponse](https://github.com/badges/shields/blob/8fcc13d5bced23f53c9f075e51b419060f6cc124/lib/error-helper.js#L8)
-   function and a badge with an appropriate status is returned: "inaccessible"
-   ![](https://img.shields.io/badge/docker_build-inaccessible-red.svg), "not found"
-   ![](https://img.shields.io/badge/docker_build-not_found-lightgrey.svg)
-   or "invalid" ![](https://img.shields.io/badge/docker_build-invalid-lightgrey.svg).
-8. The data returned by the API as JSON is parsed.
-9. Based on the result, the text and the color of the badge are altered.
-   ![](https://img.shields.io/docker/automated/jrottenberg/ffmpeg.svg)
-   ![](https://img.shields.io/docker/automated/codersos/ubuntu-remix.svg)
-10. In case of an error, an "invalid" badge is constructed.
-    ![](https://img.shields.io/badge/docker_build-invalid-lightgrey.svg)
-
-The pattern described can be found in may other badges.
-When you look at the [server.js][server], you find many other badges.
-Some of them may query the same API.
-Those can be of additional help when implementing your badge.
-
-### (4.2) Querying an API with Authentication
-
-TODO
-This has something to do with [private/secret.json](https://github.com/badges/shields/search?utf8=%E2%9C%93&q=private%2Fsecret.json&type=).
-- Credentials should be stored there.
-- You do not need to create this file, the server works without.
-- Somewhere the format (JSON? Keys?) should be documented.
-  - How to get the keys?
-
-### (4.3) Add Badges to Front Page
-
-Once you are done implementing your badge, you can add it to the collection on [shields.io](http://shields.io/).
-
-First, we make it visible locally at [http://[::1]:8080/][home]. Edit
-[lib/all-badge-examples.js][allbadges] in the right category (Build,
-Downloads, ...) and add your badge:
+Description of the code:
+1. As with the first example, we declare strict mode at the start of each file.
+2. Our badge will query a JSON API so we will extend `BaseJsonService` instead of `BaseService`. This contains some helpers to reduce the need for boilerplate when calling a JSON API.
+3. In this case we are making a version badge, which is a common patten. Instead of directly returning an object in this badge we will use a helper function to format our data consistently. There are a variety of helper functions to help with common tasks in `/lib`.
```

Comment:
I'd love to be able to link to some useful docs here, but we don't have any (yet). In lieu of a proper internal API reference, is there something more useful we can say here than "there's a bunch of stuff in `/lib` - good luck finding it ;)"

Reply:
I'd suggest we survey the few functions that the existing new-style services are using, and list them. You mention renderVersionBadge but we could add e.g. renderLicenseBadge, metric, and whatever else the new-style badges are using now.